app/controllers/search_module.js

```diff
@@ -0,0 +1,92 @@
+import Ember from "ember";
+import InfinityRoute from "ember-infinity/mixins/route";
+
+export default Ember.Controller.extend(InfinityRoute, {
+
+  queryParams: ["showQuantityItems"],
+  showQuantityItems: false,
+
+  getCurrentUser: Ember.computed(function(){
+    var store = this.get('store');
+    var currentUser = store.peekAll('user_profile').get('firstObject') || null;
+    return currentUser;
+  }).volatile(),
+
+  sanitizeString(str){
+    str = str.replace(/[^a-z0-9áéíóúñü \.,_-]/gim,"");
+    return str.trim();
+  },
+
+  searchText: Ember.computed('searchInput',{
+    get() {
+      return this.get('searchInput') || "";
+    },
+
+    set(key, value) {
+      return this.sanitizeString(value);
+    }
+  }),
+
+  i18n: Ember.inject.service(),
+  store: Ember.inject.service(),
+  isLoading: false,
+  hasNoResults: false,
+  itemSetId: null,
+  unloadAll: false,
+  minSearchTextLength: 0,
+  searchInput: null,
+  toDesignateItem: null,
+
+  hasSearchText: Ember.computed("searchText", function() {
+    return !!this.get("searchText");
+  }),
+
+  onSearchTextChange: Ember.observer("searchText", function() {
+    // wait before applying the filter
+    if (this.get("searchText").length > this.get("minSearchTextLength")) {
+      this.set("itemSetId", null);
+      Ember.run.debounce(this, this.applyFilter, 500);
+    } else {
+      this.set("filteredResults", []);
+    }
+  }),
+
+  applyFilter() {
+    var searchText = this.get("searchText");
+    if (searchText.length > 0) {
+      this.set("isLoading", true);
+      this.set("hasNoResults", false);
+      if(this.get("unloadAll")) { this.get("store").unloadAll(); }
+
+      this.infinityModel(this.get("searchModelName"),
```

Comment:
@abulsayyad123 Did you get a change to test this infinite scrolling ? 
It seemed to have problems when I tested it on Stock. 
A good way to test is to make smaller pages (e.g 5) and make sure we can scroll to see everything. It would be interesting to see how it behaves here : )

Reply:
@abulsayyad123 , I see infiniteScroll here but I don't see `"ember-infinity"` addon which is needed for infinite scroll. May be we should move `"ember-infinity"` also to `goodcity-lib` with `search_module` controller?

CC: @patrixr

Reply:
That's a good idea